### DISCUSSION TITLE
Byte array

### DISCUSSION THREAD

**Discussion by leoP0** [kenjdavidson/react-native-bluetooth-classic#104] [General]
How can I send a Byte array? 
Do need to set the encoding to something else besides ASCII?

Currently when I send a byte array I get the array including the comas. For example the byteArray contains [ 1, 1, 254 ]. These are sent as 1,1,254 including the commas

**Comment by kenjdavidson** [comment]
There are a couple of issues and discussions already open on doing so.  Since it was never the main goal of the original library (nor the re-implementation here) I don't have much documentation - nor have I even bothered to try and play around with it (sorry).   Essentially, without customizing your own `BluetoothConnection` being used, it's ALWAYS going to attempt to send String encoded data.

https://github.com/kenjdavidson/react-native-bluetooth-classic/discussions/85

You can look through the closed issues and see if something ends up working for you.

I also have a quick _how to_ which is more of a _if I were going to try it, this is what I'd do_ which you can run through.

https://kenjdavidson.com/react-native-bluetooth-classic/guides/hex-encoding-decoding/

But that's about it that I have available.  Good luck!  Feel free to post back if you have it working and I'll update the docs with examples, or pull requests for anything that might be worth while for others.

Feel free to go through the code and experiment with `ByteArrayDeviceConnectionImpl`.  I attempted to start playing with it at one point, but then Covid happened and more important things came up.  There should be enough information to make due.

https://github.com/kenjdavidson/react-native-bluetooth-classic/blob/master/android/src/main/java/kjd/reactnative/bluetooth/conn/ByteArrayDeviceConnectionImpl.java
